src/core/utils/helpers/list-mapper.ts

```diff
@@ -0,0 +1,161 @@
+        ", "
+      )}`;
+    } else {
+      returnContentString = `${texts.materialByAuthorText} ${creators
+        .slice(0, -1)
+        .join(", ")} ${texts.materialAndAuthorText} ${creators.slice(-1)}`;
+    }
+  }
+  return returnContentString;
+};
+
+//
+function getYearFromDataString(date?: string) {
+  if (date) {
+    return new Date(date).getFullYear();
+  }
+  return "";
+}
+
+// Loan is a loan from Publizon, and is the equivalent
+// to the LoanV2 type in FBS. These are mapped to the same
+// so digital/physical loans/reservations can use the same components,
+// as their UI is often quite similar
+export const mapPublizonLoanToLoanType = (list: Loan[]): LoanType[] => {
+  return list.map(({ loanExpireDateUtc, orderDateUtc, libraryBook }) => {
+    return {
+      dueDate: loanExpireDateUtc,
+      loanDate: orderDateUtc,
+      isRenewable: false,
+      materialItemNumber: libraryBook?.identifier || "",
+      renewalStatusList: [],
+      loanType: null,
+      identifier: libraryBook?.identifier || null,
+      faust: null
+    };
+  });
+};
+
+// LoanV2 is a loan from FBS, and is the equivalent
+// to the Loan type in Publizon. These are mapped to the same
+// so digital/physical loans/reservations can use the same components,
+// as their UI is often quite similar
+export const mapFBSLoanToLoanType = (list: LoanV2[]): LoanType[] => {
+  return list.map(({ loanDetails, isRenewable, renewalStatusList }) => {
+    return {
+      dueDate: loanDetails.dueDate,
+      loanDate: loanDetails.loanDate,
+      renewalStatusList,
+      isRenewable,
+      materialItemNumber: loanDetails.materialItemNumber,
+      loanType: loanDetails.loanType,
+      identifier: null,
+      faust: (loanDetails.recordId as FaustId) || null
+    };
+  });
+};
+
+// RenewedLoanV2 is a renewed loan from FBS, and for the moment has no
+// equivalent in publizon. It is mapped to loan type, because the only
+// thing relevant in the UI is that the loan has been renewed, and is
+// Not renewable any longer.
+export const mapFBSRenewedLoanToLoanType = (
+  list: RenewedLoanV2[]
+): LoanType[] => {
+  return list.map(({ loanDetails }) => {
+    return {
+      dueDate: loanDetails.dueDate,
+      loanDate: loanDetails.loanDate,
+      renewalStatusList: [],
+      isRenewable: false,
+      materialItemNumber: loanDetails.materialItemNumber,
+      loanType: loanDetails.loanType,
+      identifier: null,
+      faust: (loanDetails.recordId as FaustId) || null
+    };
+  });
+};
+
+// Product is a material from Publizon, and is the equivalent
+// to a manifestation from FBI. These are mapped to the same
+// so digital/physical loans/reservations can use the same components,
+// as their UI is often quite similar
+export const mapProductToBasicDetailsType = (material: Product) => {
+  const {
+    publicationDate,
+    title,
+    description,
+    productType,
+    contributors,
+    externalProductId
+  } = material;
+
+  const {
+    text: { data: texts }
```

Comment:
I think we should avoid using the store directly if we can. We have one system for handling text strings: useText. In similar functions to this we have simply added a parameter to the function called: `t`. See: https://github.com/reload/dpl-react/blob/storybook-debug/src/core/utils/helpers/general.ts#L50:L50

Reply:
I've changed it. I used the store like that, because I saw it in a pr and assumed it was fine. But sending the hook with is fine as well, I just wasnt aware.

Reply:
👍